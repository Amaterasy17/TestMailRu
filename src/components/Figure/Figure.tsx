import React from 'react'
import PropTypes from 'prop-types'
import {Loader} from "../Loader/Loader";
import {createUseStyles} from "react-jss";

const image = createUseStyles({
    image: {
        visibility: 'hidden',
    }
})


export interface FigureProps {
    src: string,
    arWidth?: number,
    arHeight?: number,
}

export const Figure = ({src, arWidth = 16, arHeight = 9}: FigureProps) => {
    const Loading = () => {
        setLoading(false);

    }
    const [loading, setLoading] = React.useState(true);
    const [visible, setVisible] = React.useState(false);

    const imageStyle = image();

    React.useEffect(() => {


    }, [loading]);

    return (
        <>
            <div>
                {loading && <Loader/>}
                <img src={src} alt={'упс'} onLoad={Loading}/>
            </div>
        </>
    );
}


