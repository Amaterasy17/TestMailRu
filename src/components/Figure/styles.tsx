import styled, { css } from 'styled-components'

type StyledImgProps = {
    isNone: boolean,
}

type StyledImage = {
    arWidth: number,
    arHeight: number,
}

export const FigureContainer = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	width: 100%;
	box-sizing: border-box;
	height: 100%;
	overflow: hidden;
`

export const ImageContainer = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	background-color: #1f2128;
	align-items: center;
	position: relative;
	width: 100%;
	height: 100%;
	padding-top: calc(
		${(props: StyledImage) => props.arHeight / props.arWidth} * 100%
	);
`

export const Image = styled.img`
	width: 100%;
	height: 100%;
	display: block;
	object-fit: cover;
	position: absolute;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	${({ isNone }: StyledImgProps) =>
    isNone &&
    css`
			display: none;
		`}
`

export const Figcaption = styled.figcaption`
	//@font-face {
	//	font-family: Montserrat-Light;
	//	src: url('/../styles/fonts/Montserrat-Light.ttf');
	//}
	//@font-face {
	//	font-family: 'Montserrat-Regular';
	//	src: url('/../styles/fonts/Montserrat-Regular.ttf');
	//}
	font-family: -apple-system, 'Roboto', 'Oxygen', sans-serif;
	font-size: 2.5vh;
	margin-top: 3px;
	text-align: center;
`
