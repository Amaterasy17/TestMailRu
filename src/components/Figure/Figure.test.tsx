import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Figure } from './Figure';
import styled from 'styled-components';

configure({ adapter: new Adapter() });

type testProps = {
	width: number,
	height: number,
}

const TestDiv = styled.div`
	width: ${(props: testProps) => `${props.width}px`};
	height: ${(props: testProps) => `${props.height}px`};
`

describe('Tests of Figure component', () => {
			test.each([
				[300, 500, 5, 3, 'https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006'],
				[300, 500, 2, 1, 'https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006'],
				[300, 500, 5, 1, 'https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006'],
				[300, 100, 16, 9, 'https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006'],
				[300, 500, 4, 3, 'https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006']
			])('tests of resizing pictures', (width: number, height: number, arHeight: number, arWidth:number, src: string) => {
				const figure = shallow(
					<TestDiv width={width} height={height}>
						<Figure
							src={src}
							arWidth={arWidth}
							arHeight={arHeight}
						/>
					</TestDiv>,
				);
				expect(figure).toMatchSnapshot();
			});
			it('test with not valid src of photo', () => {
				const figure = shallow(
					<TestDiv width={300} height={500}>
						<Figure
							src='fakePicture'
							arWidth={9}
							arHeight={16}
						/>
					</TestDiv>,
				);
				expect(figure).toMatchSnapshot();
			});

			it('default with figcaption', () => {
				const figure = shallow(
					<TestDiv width={300} height={300}>
						<Figure
							src='https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006'
						>
							Автор:{' '}
							<a href="https://www.pexels.com/@simonmigaj">Simon Migaj</a>
						</Figure>
					</TestDiv>,
				);
				expect(figure).toMatchSnapshot();
			});
})