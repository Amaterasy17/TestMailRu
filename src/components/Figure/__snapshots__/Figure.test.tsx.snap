// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of Figure component default with figcaption 1`] = `
<div
  className="sc-fujyAs cwgfdV"
  height={300}
  width={300}
>
  <Figure
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  >
    Автор:
     
    <a
      href="https://www.pexels.com/@simonmigaj"
    >
      Simon Migaj
    </a>
  </Figure>
</div>
`;

exports[`Tests of Figure component test with not valid src of photo 1`] = `
<div
  className="sc-fujyAs cweGFr"
  height={500}
  width={300}
>
  <Figure
    arHeight={16}
    arWidth={9}
    src="fakePicture"
  />
</div>
`;

exports[`Tests of Figure component tests of resizing pictures 1`] = `
<div
  className="sc-fujyAs cweGFr"
  height={500}
  width={300}
>
  <Figure
    arHeight={5}
    arWidth={3}
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  />
</div>
`;

exports[`Tests of Figure component tests of resizing pictures 2`] = `
<div
  className="sc-fujyAs cweGFr"
  height={500}
  width={300}
>
  <Figure
    arHeight={2}
    arWidth={1}
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  />
</div>
`;

exports[`Tests of Figure component tests of resizing pictures 3`] = `
<div
  className="sc-fujyAs cweGFr"
  height={500}
  width={300}
>
  <Figure
    arHeight={5}
    arWidth={1}
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  />
</div>
`;

exports[`Tests of Figure component tests of resizing pictures 4`] = `
<div
  className="sc-fujyAs cwfECj"
  height={100}
  width={300}
>
  <Figure
    arHeight={16}
    arWidth={9}
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  />
</div>
`;

exports[`Tests of Figure component tests of resizing pictures 5`] = `
<div
  className="sc-fujyAs cweGFr"
  height={500}
  width={300}
>
  <Figure
    arHeight={4}
    arWidth={3}
    src="https://images.pexels.com/photos/1647214/pexels-photo-1647214.jpeg?dl&fit=crop&crop=entropy&w=640&h=1006"
  />
</div>
`;
